atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/Token.java

```diff
@@ -19,6 +19,8 @@
 import org.immutables.value.Value;
 
 @Value.Immutable
+@Value.Style(visibility = Value.Style.ImplementationVisibility.PACKAGE)
+@SuppressWarnings("ClassInitializationDeadlock")
 abstract class Token {
     @Nullable
```

Comment:
Instead of breaking the interface contract by moving the constant, we are setting the impl visibility to PACKAGE (this is done to avoid initializing the immutable first) and suppressing the error.

Reply:
setting the implementation visibility only helps avoid external usages which could potentially cause a deadlock and within the package we have to be sensible while using the immutable. 
The check is still right in this case to break, which is why we suppress it.
This cannot be done in the cases where the subclass has usages in other packages. In those cases we move out the constants to a separate util. To have a separate a util only for a constant seems a but janky which is why we have take the first approach if possible.

Reply:
If we try to initialise ImmutableClass before Class, that will require us to initialise Class, but because of the static constant, that requires an initialisation of ImmutableClass and this deadlocks because there is a lock on ImmutableClass.

If you ensure that Class gets initialised first, then everything works fine, so by making access to ImmutableClass package private (to prevent external badness) and exposing a builder on the Class interface to be used instead of the ImmutableClass builder directly, we can avoid this issue

Reply:
Package private may still be accessed externally. I considered special casing in the errorprone rule, but it would require additional validation that nothing accesses package-private generated immutables outside of the defining class, which we've seen more frequently than one would hope.